Xamarin.Essentials/MediaPicker/MediaPicker.android.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Android.App;
+using Android.Content;
+using Android.Provider;
+
+namespace Xamarin.Essentials
+{
+    public static partial class MediaPicker
+    {
+        enum RequestCode
+        {
+            Base = 10010,
+
+            PickPhoto = Base + 1,
+            TakePhoto = Base + 2,
+            PickVideo = Base + 3,
+            TakeVideo = Base + 4,
+
+            Min = PickPhoto,
+            Max = TakeVideo,
+        }
+
+        static MediaPicker()
+        {
+            Platform.ActivityResult += OnActivityResult;
+        }
+
+        static void OnActivityResult(int requestCode, Result resultCode, Intent data)
+        {
+            // this wasn't us
+            if (requestCode < (int)RequestCode.Min || requestCode > (int)RequestCode.Max)
+                return;
+
+            // the user canceled
+            if (resultCode == Result.Canceled)
+            {
+                OnMediaPicked(new MediaPickedEventArgs(true));
+                return;
+            }
+
+            // an OK, so process
+            var args = ProcessPickerIntent(requestCode, data);
+
+            // raise the global event
+            OnMediaPicked(args);
+        }
+
+        static async Task<MediaFile> PlatformShowPhotoPickerAsync(MediaPickerOptions options)
+        {
+            // make sure we have permission and an activity
+            await Permissions.RequireAsync(PermissionType.ExternalStorage);
+
+            var intent = new Intent(Intent.ActionPick);
+            intent.SetType("image/*");
+
+            try
+            {
+                // launch the picker intent via the intermediate activity
+                var data = await IntermediateActivity.StartAsync(intent, (int)RequestCode.PickPhoto);
```

Comment:
We can use a `Task` to get the result of an activity. This should be OK as the task will complete when the picker activity completes. We may have to investigate what happens if the activity is killed and then the task resumes. There may be some mono/managed code that will run as part of the completing async machine.

Reply:
I still think it might be worth having the Intermediate Activity have the _option_ to start a foreground service for the life of itself so that Android absolutely will _not_ kill the activity.  The foreground service needs a persistent notification which could just have the current app's name.  The notification will go away as soon as the intermediate activity is finished.  We could gate this behaviour with a platform specific flag (`MediaPicker.KeepAliveWithForegroundService`) or something.